index.js

```diff
@@ -188,37 +150,39 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
   }
 
   function getAddedSubOrgConfigName(payload) {
-    const repoSettingPattern = new Glob(".github/suborgs/*.yml")
+    const settingPattern = new Glob(".github/suborgs/*.yml")
 
     let commit = payload.commits.find(c => {
       return ( c.added.find(s => {
         robot.log.debug(JSON.stringify(s))
-        return ( s.search(repoSettingPattern)>=0 )
+        return ( s.search(settingPattern)>=0 )
       }) !== undefined )
     })
 
     if (commit) {
-      robot.log.debug(`${JSON.stringify(commit)}`)
-      return {suborg: commit.added[0].match(repoSettingPattern)[1], org: payload.repository.owner.name}
+      const matches = commit.added[0].match(settingPattern)
+      robot.log.debug(`${JSON.stringify(commit)} \n ${matches[1]}`)
+      return { name: matches[1]+".yml", path: commit.added[0] }
     } else {
       robot.log.debug(`No additions to suborgs configs`)
     }
     return undefined
   }
 
   function getModifiedSubOrgConfigName(payload) {
-    const repoSettingPattern = new Glob(".github/suborgs/*.yml")
+    const settingPattern = new Glob(".github/suborgs/*.yml")
 
     let commit = payload.commits.find(c => {
       return ( c.modified.find(s => {
         robot.log.debug(JSON.stringify(s))
-        return ( s.search(repoSettingPattern)>=0 )
+        return ( s.search(settingPattern)>=0 )
       }) !== undefined )
     })
 
     if (commit) {
-      robot.log.debug(`${JSON.stringify(commit)} \n ${commit.modified[0].match(repoSettingPattern)[1]}`)
-      return repo = {suborg: commit.modified[0].match(repoSettingPattern)[1], org: payload.repository.owner.name}
+      const matches = commit.modified[0].match(settingPattern)
+      robot.log.debug(`${JSON.stringify(commit)} \n ${matches[1]}`)
```

Comment:
If there is not match, then `matches[1]` will crash.

This may happen if you have in the same commit both a `settings.yml` and at least one suborg because find is getting any commit that contains one suborg (but it may contain other files as well that are not suborgs).

If the first file on that commit is not a suborg this will fail

Reply:
This is still an unresolved issue. I have taken a technical debt to figure this out later when we have a good handle on the scale issue.